payjoin/src/receive/v2/mod.rs

```diff
@@ -533,7 +533,11 @@ impl PayjoinProposal {
             target_resource.as_str(),
             Some(&body),
         )?;
-        let url = ohttp_relay.into_url().map_err(InternalSessionError::ParseUrl)?;
+
+        let relay_base = ohttp_relay.into_url().map_err(InternalSessionError::ParseUrl)?;
+        let url = relay_base
+            .join(self.context.directory.as_str())
+            .map_err(|e| InternalSessionError::ParseUrl(crate::into_url::Error::ParseError(e)))?;
```

Comment:
maybe we could have from impls? I don't see these as a problem but the thought came over me.

Reply:
Agreed, the explicit InternalSessionError::ParseUrl wrapping the error makes more sense here. I'll check out the SystemTime for Error impl and see if something explicit where that's used makes more sense too.